config/metric.go

```diff
@@ -319,7 +313,23 @@ func prometheusReader(ctx context.Context, prometheusConfig *Prometheus) (sdkmet
 		if err != nil {
 			return nil, err
 		}
-		otelprom.WithResourceAsConstantLabels(f)
+		opts = append(opts, otelprom.WithResourceAsConstantLabels(f))
+	}
+
+	return opts, nil
+}
+
+func prometheusReader(ctx context.Context, prometheusConfig *Prometheus) (sdkmetric.Reader, error) {
```

Comment:
nit: I would move the `prometheusReader` function before `prometheusReaderOpts`.

Reply:
updated